programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -546,13 +541,14 @@ declare_syscall!(
                 return Ok(0);
```

Comment:
Could the old invoke_context.get_allocator()? actually fail? Because if so, now
this potentially succeeds for some inputs that previously might have failed